openmc/mgxs/mgxs.py

```diff
@@ -824,9 +824,9 @@ def get_subdomain_avg_xs(self, subdomains='all'):
             mean = tally.get_reshaped_data(value='mean')
             std_dev = tally.get_reshaped_data(value='std_dev')
 
-            # Get the mean of the mean, std. dev. across requested subdomains
-            mean = np.mean(mean[subdomains, ...], axis=0)
-            std_dev = np.mean(std_dev[subdomains, ...]**2, axis=0)
+            # Get the mean, std. dev. across requested subdomains
+            mean = np.sum(mean[subdomains, ...], axis=0)
+            std_dev = np.sum(std_dev[subdomains, ...]**2, axis=0)
```

Comment:
The earlier version computed the geometric average of the flux and reaction
rate in each energy group and subdomain. Then it used these geometric
averages to compute the subdomain-averaged MGXS in each energy group.

The new version sums the flux and reaction rate tallies across subdomains
to compute the aggregate. This is effectively the flux weighted average of
the MGXS in each energy group across subdomains. One way to think about
this is that those subdomains with little flux are weighted less in the
average than those subdomains with more flux. This is exactly what is done
for a cell-based tally where the cell is replicated throughout a geometry.
On Nov 9, 2015 9:02 AM, "Paul Romano" <notifications@github.com> wrote:

> In openmc/mgxs/mgxs.py
> <https://github.com/mit-crpg/openmc/pull/496#discussion_r44276963>:
>
> > @@ -824,9 +824,9 @@ def get_subdomain_avg_xs(self, subdomains='all'):
> >              mean = tally.get_reshaped_data(value='mean')
> >              std_dev = tally.get_reshaped_data(value='std_dev')
> >
> > -            # Get the mean of the mean, std. dev. across requested subdomains
> > -            mean = np.mean(mean[subdomains, ...], axis=0)
> > -            std_dev = np.mean(std_dev[subdomains, ...]**2, axis=0)
> > +            # Get the mean, std. dev. across requested subdomains
> > +            mean = np.sum(mean[subdomains, ...], axis=0)
> > +            std_dev = np.sum(std_dev[subdomains, ...]**2, axis=0)
>
> I don't quite understand how this change reflects what you described in
> words. Can you elaborate?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/496/files#r44276963>.
>